src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -10,21 +10,58 @@
 
     Ultrasonic ultrasonicSensor;
 
-    UltrasonicPoller(Ultrasonic sensor, int time) {
+    // Consider renaming, couldn't think of a better one ATM
+    boolean noBalls;
+    long time;
+    double distance;
+    double currentRange;
+
+    long startTime;
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance, long rate) {
         // sensor is pretty self explanatory
         // time is how long you want the sensor to not detect an object until true is returned
-        super(25, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
+        // distance is the minimum distance threshold
+        super(rate, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
         this.ultrasonicSensor = sensor;
+        this.time = time;
+        this.distance = distance;
+        this.noBalls = false;
+        this.startTime = 0;
+
+    }
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance) {
+        this(sensor, time, distance, 25);
     }
 
     @Override
     public void defineResources(){
         require(ultrasonicSensor);
     }
 
+    public void setTime(long time) {
+        this.time = time;
+    }
+
+    public void setDistance(double distance){
+        this.distance = distance;
+    }
+
+    // Consider renaming, probably want to generalize the name if we ever re-use it
+    public boolean checkUnloaded(){
+        return this.noBalls;
+    }
+
     @Override
     public void task(){
-        // Placholder
-        return;
+        this.currentRange = this.ultrasonicSensor.getRangeMM();
+        this.startTime = System.nanoTime();
+        if((this.currentRange > this.distance) && (((System.nanoTime() - this.startTime) / 100000) > this.time)) {
```

Comment:
Your two calls to `System.nanoTime()` happen back-to-back, so there's almost certainly no chance that the time threshold will have elapsed by the second evaluation. You need to store the `startTime` timestamp only on the "edges" of the state transition - currently you're setting it every time the task evaluates.

```
[X]  [X]  [ ]  [ ]  [ ]  [ ]      [ ]  [ ]  [ ]  [X]
           ^                 ...   ^              ^
          mark                   threshold       reset
```

Reply:
Yeah that's true. I made some modifications and didn't stop to reconsider the implications of them.

Reply:
Lol wait, I just realized I made a major logic error. What I really want to do is only get the start time on the first run of the task (that's probably what I intended last night, I just wasn't thinking), then compare the difference and reset the start time only if a ball is detected.

Reply:
The reason this happened is because I noticed that because startTime has an initial value of 0, it will always set noBalls to true (because System.nanoTime - 0 = System.nanotime, which will be much bigger than whatever the time threshold is), so I said: "Ok, let's set the start time to avoid this edge case from happening", so I set the start time at the beginning of the task. What I neglected is that the task is keep on looping while it's enabled, so I didn't think about how the time will get reset on every run, regardless of what's happening.

Reply:
The two commits I just made should resolve this issue, but I feel as if the way I went about it was, although perfectly functional and doesn't hit performance at all (literally just a single comparison), horrible stylistically, and too 'bloated'.